### ISSUE TITLE
Consider herokuish to use in slugrunner and slugbuilder

### ISSUE THREAD

**Issue by saulshanabrook** [flynn/flynn#758]
Would it make sense to use [gliderlabs/herokuish](https://github.com/gliderlabs/herokuish) in the slugrunner and slugbuilder projects?

It seems advantageous to me because then flynn would share some of the same building blocks as [buildstep](https://github.com/progrium/buildstep) (once progrium/buildstep#129 merged) to reduce code complexity and increase stability.

Does anyone have thoughts on whether this would make sense?

**Comment by titanous** [comment]
We're big fans of using stuff that already exists. Unfortunately the Glider Labs team only started on herokuish a couple of months ago while we implemented Heroku compatibility in Flynn over a year earlier. It's hard to justify replacing Flynn's existing stable and fully tested implementation for a variety of reasons:

 - Heroku compatibility isn't a particularly hard problem and doesn't make up much of our codebase or development time (in fact most existing platform projects have implemented Heroku compatibility on their own fairly trivially). Heroku compatibility also isn't really a moving target so there isn't really a need to stay "up to date".
 
 - We've already customized our implementation and expect that with time the level of customization will only increase. If we relied on an external component like herokuish there would likely be additional overhead caused by trying to harmonize our changing needs with those of other users of the project. This could easily cause one or more users to resort to maintaining a fork specific to their needs, defeating the point of a shared repo.
 
 - When we do replace the existing implementation in Flynn it will be with something in pure Go. Herokuish is mostly bash, so that doesn't help us.
 
Basically there would be no increase to stability and isn't much if anything to be gained but there would be a cost to switching, a higher maintenance cost, and a high likelyhood of replacing it again in the not-distant future.
